### PR TITLE
Sassd

### PR DESCRIPTION
**Pull request by wenbo-yu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
We added points_in_boxes_with_offsets_cpu_forward under /ops. The purpose is in addition to get the box indexes of each points, it also return the offsets (distance to the box center). The purpose is that we want to add a new model in mmdet3d. This function is crucial to calculate the loss. mmdet3dd suggested us to add this to mmcv.

## Modification
points_in_boxes_with_offsets_cpu_forward related functions under ops. We don't know the purpose of the file under ops/csrc/parrots. Though we implement the functions referring to the others.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by zhouzaida** [comment]
Hi @wenbo-yu , thanks for your contribution. The lint failed and you can refer to https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md#python to install pre-commit to automatically fix the lint problem.

**Comment by filaPro** [comment]
Just out of curiosity, why not to call `points_in_boxes` and then calculate offsets in `pytorch`? Will it be noticeably slower?

File: mmcv/ops/points_in_boxes.py [hunk 1]
```diff
@@ -94,6 +94,51 @@ def points_in_boxes_cpu(points, boxes):
     return point_indices
 
 
+def points_in_boxes_with_offsets_cpu(points, boxes):
```
Comment (AllentDan) [inline mmcv/ops/points_in_boxes.py:97]: may add type hint here

**Review by AllentDan — COMMENTED** [review]
Please fix the lint error and add a unit test. We also need to update those ops in https://github.com/open-mmlab/mmcv/blob/master/docs/en/understand_mmcv/ops.md and https://github.com/open-mmlab/mmcv/blob/master/docs/zh_cn/understand_mmcv/ops.md

**Pull request closed without merging**